extensions/typescript-language-features/src/tsServer/webServer.ts

```diff
@@ -0,0 +1,540 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+/// <reference types='typescript/lib/tsserverlibrary' />
+/// <reference lib='webworker.importscripts' />
+/// <reference lib='dom' />
+// BEGIN misc internals
+const hasArgument: (argumentName: string) => boolean = (ts as any).server.hasArgument;
+const findArgument: (argumentName: string) => string | undefined = (ts as any).server.findArgument;
+const nowString: () => string = (ts as any).server.nowString;
+const noop = () => { };
+const perfLogger = {
+	logEvent: noop,
+	logErrEvent(_: any) { },
+	logPerfEvent(_: any) { },
+	logInfoEvent(_: any) { },
+	logStartCommand: noop,
+	logStopCommand: noop,
+	logStartUpdateProgram: noop,
+	logStopUpdateProgram: noop,
+	logStartUpdateGraph: noop,
+	logStopUpdateGraph: noop,
+	logStartResolveModule: noop,
+	logStopResolveModule: noop,
+	logStartParseSourceFile: noop,
+	logStopParseSourceFile: noop,
+	logStartReadFile: noop,
+	logStopReadFile: noop,
+	logStartBindFile: noop,
+	logStopBindFile: noop,
+	logStartScheduledOperation: noop,
+	logStopScheduledOperation: noop,
+};
+const assertNever: (member: never) => never = (ts as any).Debug.assertNever;
+const memoize: <T>(callback: () => T) => () => T = (ts as any).memoize;
+const ensureTrailingDirectorySeparator: (path: string) => string = (ts as any).ensureTrailingDirectorySeparator;
+const getDirectoryPath: (path: string) => string = (ts as any).getDirectoryPath;
+const directorySeparator: string = (ts as any).directorySeparator;
+const combinePaths: (path: string, ...paths: (string | undefined)[]) => string = (ts as any).combinePaths;
+// NOTE: This relies on the current way that vscode injects dynamicImport into ts.server and will probably change
+const dynamicImport: ((id: string) => Promise<any>) | undefined = (ts as any).server.dynamicImport;
```

Comment:
This doesn't exist on `tsserverlibrary` as it was solely a detail of `tsserver`. You might have to do something special to get the import expression here instead.

Reply:
Assuming that we still ship dynamicImportCompat.js, it's injected at the very start of tsserver.web.js by the webpack hook: 

```js
var ts;
(function (ts) {
    var server;
    (function (server) {
        server.dynamicImport = (id) => import(id);
    })(server = ts.server || (ts.server = {}));
})(ts || (ts = {}));
```

However, I think we have plans to include it tsserverlibrary itself somewhere so this prefix isn't needed. Or maybe work around needing it at all.

Reply:
dynamicImport is actually defined in the web server code, so would have to be in the code here.